### PR TITLE
fix(nix): update path to hydra build for 2.24 and newer

### PR DESCRIPTION
**Pull request by sandydoo**
Renovate currently fails to download Nix because the URL for the latest version has been updated.

```
WARN: Error updating flake.lock (branch="renovate/lock-file-maintenance")
{
  "err": {
    "cmd": "/bin/sh -c install-tool nix 2.24.1",
    "stderr": "",
    "stdout": "installing v2 tool nix v2.24\n[12:53:34.669] INFO (53): Installing tool nix@2.24.1...\nDownload failed: https://hydra.nixos.org/job/nix/maintenance-2.24/buildStatic.x86_64-linux/latest/download-by-type/file/binary-dist\nDownload failed, retrying\nDownload failed: https://hydra.nixos.org/job/nix/maintenance-2.24/buildStatic.x86_64-linux/latest/download-by-type/file/binary-dist\nDownload failed, retrying\nDownload failed: https://hydra.nixos.org/job/nix/maintenance-2.24/buildStatic.x86_64-linux/latest/download-by-type/file/binary-dist\nDownload failed: https://hydra.nixos.org/job/nix/maintenance-2.24/buildStatic.x86_64-linux/latest/download-by-type/file/binary-dist\n[12:53:37.663] INFO (103): Downloading file ...\n    url: \"https://hydra.nixos.org/job/nix/maintenance-2.24/buildStatic.x86_64-linux/latest/download-by-type/file/binary-dist\"\n    output: \"/tmp/renovate/cache/containerbase/50789f3416da6a7a362045e5add6d2ba81fef302b1678eae91ab63978e430317/binary-dist\"\n[12:53:38.117] ERROR (103): Response code 404 (Not Found)\n[12:53:38.118] FATAL (103): Download failed in 454ms.\n[12:53:38.296] ERROR (53): Command failed with exit code 1: /usr/local/containerbase/bin/install-tool.sh nix 2.24.1\n[12:53:38.297] FATAL (53): Install tool nix failed in 3.6s.\n",
```

Changes:

- Bumped the version of Nix used in tests to the latest. This replicates the issue.
- Updated the Hydra URL for Nix v2.24+.
- Made a note in the registry docs about the URL change.

### CODE REVIEW

**Comment by BenoitKnecht** [comment]
@sandydoo Hell of a timing: #3067. :sweat_smile:

File: src/usr/local/containerbase/tools/v2/nix.sh [hunk 1]
```diff
@@ -9,14 +9,22 @@ function install_tool() {
...
 
   # if [[ ${MAJOR} -lt 2 || (${MAJOR} -eq 2 && ${MINOR} -lt 14) ]]; then
   #   echo "Nix version ${TOOL_VERSION} is not supported! Use v2.14 or higher." >&2
   #   exit 1
   # fi
 
   arch=$(uname -m)
-  file=$(get_from_url "https://hydra.nixos.org/job/nix/maintenance-${TOOL_VERSION}/buildStatic.${arch}-linux/latest/download-by-type/file/binary-dist")
+
+  if [[ ${MAJOR} -gt 2 || (${MAJOR} -eq 2 && ${MINOR} -lt 24) ]]; then
```
Comment (BenoitKnecht) [inline src/usr/local/containerbase/tools/v2/nix.sh:21]: I think you meant ` ${MAJOR} -lt 2` as the first condition.

**Comment by sandydoo** [comment]
@BenoitKnecht, we should've had a battle of the PRs! suffix vs path

**Comment by BenoitKnecht** [comment]
> @BenoitKnecht, we should've had a battle of the PRs! suffix vs path

@sandydoo Don't tempt me. :crossed_swords: :smile:

File: test/Dockerfile.jammy [hunk 2]
```diff
@@ -77,7 +77,7 @@ RUN install-tool kubectl v1.30.3
 RUN install-tool kustomize 5.4.3
 
 # renovate: datasource=github-tags packageName=NixOS/nix
-RUN install-tool nix 2.23.3
+RUN install-tool nix 2.24.1
```
Comment (viceice) [inline test/Dockerfile.jammy:80]: please revert the test changes

Comment (sandydoo) [reply test/Dockerfile.jammy]: Done 👍

**Review by viceice — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**